apps/nestjs-backend/src/features/share/share.service.ts

```diff
@@ -177,4 +231,107 @@ export class ShareService {
+    const { view, tableId } = shareInfo;
+    const { fieldId } = query;
+
+    // only form view can get all records
+    if (view.type === ViewType.Form) {
+      return this.getViewFormCollaborators(shareInfo);
+    }
+
+    if (!fieldId) {
+      throw new BadRequestException('fieldId is required');
+    }
+
+    await this.preCheckFieldHidden(view, fieldId);
+
+    // user field check
+    const field = await this.fieldService.getField(tableId, fieldId);
+    // All user field, contains lastModifiedBy, createdBy
+    if (field.type !== FieldType.User) {
+      throw new ForbiddenException('field type is not user field');
+    }
+
+    return this.getViewFilterCollaborators(shareInfo, field);
+  }
+
+  private async getViewFilterUserQuery(
+    tableId: string,
+    filter: IFilter | undefined,
+    userField: IFieldVo,
+    fieldMap: Record<string, IFieldInstance>
+  ) {
+    const dbTableName = await this.recordService.getDbTableName(tableId);
+    const queryBuilder = this.knex(dbTableName);
+    const { isMultipleCellValue, dbFieldName } = userField;
+
+    this.dbProvider.shareFilterCollaboratorsQuery(queryBuilder, dbFieldName, isMultipleCellValue);
+    queryBuilder.whereNotNull(dbFieldName);
+    this.dbProvider.filterQuery(queryBuilder, fieldMap, filter).appendQueryBuilder();
+
+    return this.knex('users')
+      .select('id', 'email', 'name', 'avatar')
+      .from(this.knex.raw(`(${queryBuilder.toQuery()}) AS coll`))
+      .leftJoin('users', 'users.id', '=', 'coll.user_id')
+      .toQuery();
+  }
+
+  async getViewFilterCollaborators(shareInfo: IShareViewInfo, field: IFieldVo) {
+    const { tableId, view } = shareInfo;
+    const fields = await this.fieldService.getFieldsByQuery(tableId, {
+      viewId: view.id,
+    });
+
+    const nativeQuery = await this.getViewFilterUserQuery(
+      tableId,
+      view.filter,
+      field,
+      fields.reduce(
+        (acc, field) => {
+          acc[field.id] = createFieldInstanceByVo(field);
+          return acc;
+        },
+        {} as Record<string, IFieldInstance>
+      )
+    );
+
+    const users = await this.prismaService
+      .txClient()
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      .$queryRawUnsafe<{ id: string; email: string; name: string; avatar: string | null }[]>(
+        nativeQuery
+      );
+
+    return users.map(({ id, email, name, avatar }) => ({
+      userId: id,
+      email,
+      userName: name,
+      avatar: avatar && getFullStorageUrl(avatar),
+    }));
+  }
+
+  async getViewFormCollaborators(shareInfo: IShareViewInfo) {
+    const { tableId, view } = shareInfo;
+
+    if (view.type !== ViewType.Form) {
+      throw new ForbiddenException('view type is not allowed');
+    }
+
+    const fields = await this.fieldService.getFieldsByQuery(tableId, {
+      viewId: view.id,
+      filterHidden: !view.shareMeta?.includeHiddenField,
```

Comment:
For the share view of a form, there is no `includeHiddenField` property, so there's no need to pass in the `filterHidden` attribute